src/main/scala/no/ndla/imageapi/service/ImageConverter.scala

```diff
@@ -74,39 +74,36 @@ trait ImageConverter {
       crop(originalImage, sourceImage, topLeft, bottomRight)
     }
 
-    private def getStartEnd(size: Int, targetSize: Int, focal: Int): (Int, Int) = {
-      val ts = min(targetSize, size) / 2
-      val (start, end) = (focal - ts, focal + ts)
-      val (startRem, endRem) = (abs(min(start, 0)), max(end - size, 0))
+    private def getStartEndCoords(focalPoint: Int, targetDimensionSize: Int, originalDimensionSize: Int): (Int, Int) = {
+      val ts = min(targetDimensionSize.toDouble, originalDimensionSize.toDouble) / 2.0
+      val (start, end) = (focalPoint - ts.floor.toInt, focalPoint + ts.round.toInt)
+      val (startRem, endRem) = (abs(min(start, 0)), max(end - originalDimensionSize, 0))
 
-      (max(start - endRem, 0), min(end + startRem, size))
+      (max(start - endRem, 0), min(end + startRem, originalDimensionSize))
     }
 
-    def dynamicCrop(image: ImageStream, percentFocalPoint: PercentPoint, targetWidth: Int, targetHeight: Int): Try[ImageStream] = {
+    def dynamicCrop(image: ImageStream, percentFocalPoint: PercentPoint, targetWidthOpt: Option[Int], targetHeightOpt: Option[Int]): Try[ImageStream] = {
       val sourceImage = image.sourceImage
       val focalPoint = toPixelPoint(percentFocalPoint, sourceImage)
       val (imageWidth, imageHeight) = (sourceImage.getWidth, sourceImage.getHeight)
 
-      val (startY, endY) = getStartEnd(imageHeight, targetHeight, focalPoint.y)
-      val (startX, endX) = getStartEnd(imageWidth, targetWidth, focalPoint.x)
-
-      crop(image, sourceImage, PixelPoint(startX, startY), PixelPoint(endX, endY))
-    }
-
-    def dynamicCropWidth(image: ImageStream, percentFocalPoint: PercentPoint, targetWidth: Int): Try[ImageStream] = {
-      val (imageWidth, imageHeight) = (image.sourceImage.getWidth, image.sourceImage.getHeight)
-      val actualTargetWidth = min(imageWidth, targetWidth)
-
-      val widthReductionPercent: Double = actualTargetWidth.toDouble / imageWidth.toDouble
-      dynamicCrop(image, percentFocalPoint, actualTargetWidth, (imageHeight * widthReductionPercent).toInt)
-    }
+      val (targetWidth: Int, targetHeight: Int) = (targetWidthOpt, targetHeightOpt) match {
+        case (None, None) => return Success(image)
+        case (Some(w), Some(h)) => (w, h)
```

Comment:
Burde du ikke her også sjekke at `w` og `h` ikke er større enn hhv. `imageWidth` og `imageHeight` slik du gjør i de to andre tilfellene nedenfor?

Reply:
Egentlig trengs det ikke gjøres her fordi det gjøres i `getStartEndCoords`, men legger det til her også for lesbarhets skyld. I tilfellene nedenfor er det nødvendig for at regnestykket skal bli riktig